### PR TITLE
Assignment - Week 1

### PR DESCRIPTION
**Pull request by nduyhai**
Assignment - Week 1

### CODE REVIEW

File: cmd/main.go [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+	"strconv"
+	"strings"
+)
+
+func main() {
+	reader := bufio.NewReader(os.Stdin)
+	fmt.Println("Golang Bootcamp Shell")
+	fmt.Println("---------------------")
+
+	for {
```
Comment (havinhdat) [inline cmd/main.go:17]: Can use `reader.Scan()`
https://golang.org/pkg/bufio/#Scanner.Scan

File: cmd/main.go [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+			}
+		}
+
+	}
+
+}
+
+func handle(input string) (result string, e error) {
+
+	r := regexp.MustCompile("\\s+")
```
Comment (havinhdat) [inline cmd/main.go]: regexp will be not very efficient in complex or long input.
For simple requirement, better not over engineering.
in this case, we can use `strings.Fields`
https://golang.org/pkg/strings/#Fields

File: cmd/main.go [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+	if len(arg) == 3 {
+		a, error := strconv.Atoi(arg[0])
+		if error != nil {
+			e = errors.New("sorry, please type valid operand 1")
+		}
+
+		b, error := strconv.Atoi(arg[2])
+		if error != nil {
+			e = errors.New("sorry, please type valid operand 2")
+		}
```
Comment (havinhdat) [inline cmd/main.go:47]: parsing for arg[0] and arg[2] code are the same, better keep it DRY by separate to a function

File: cmd/main.go [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+
+		case "*":
+			result = strconv.FormatInt(int64(a*b), 10)
+
+		case "/":
+			if b == 0 {
+				e = errors.New("sorry, cannot  device to zero")
+
+			} else {
+				result = strconv.FormatFloat(float64(float64(a)/float64(b)), 'g', -1, 64)
```
Comment (havinhdat) [inline cmd/main.go:63]: `float64(a)/float64(b)` alone is sufficient

File: cmd/main.go [hunk 5]
```diff
@@ -0,0 +1,77 @@
...
+			}
+
+		default:
+			e = errors.New("sorry, please type valid operator")
```
Comment (havinhdat) [inline cmd/main.go]: duplicate error, can put it in the top or in global scope to reuse

File: cmd/main.go (hunk 2 above)
Comment (nduyhai) [reply cmd/main.go]: thanks u :D

**Pull request closed without merging**